### PR TITLE
Followers: use custom post types and postmeta to store

### PR DESCRIPTION
**Pull request by mattwiebe**
We might have trouble using `termmeta` on WordPress.com, so here is a speculative implementation of `Followers` and `Follower` that are backed by a Custom Post Type and postmeta.

Note that this should also migrate from storing the associated user in `post_author` to allowing multiple metadata `user_id` on the post so that the same Follower can be assigned to multiple users.

Also note that this isn't even tested yet

## Proposed changes:

*

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Review by mattwiebe — COMMENTED** [review]
It won't let me approve it since I started it so I'll just say that this is good to merge 🚢 

Great work taking this over the finish line! So many fiddly details.

My other notes are mostly about caching issues that I want to address but I don't see them as blockers for landing this, especially since everything else depends on it. We can open new PRs for that and any other issues we encounter in testing.

File: includes/model/class-follower.php [hunk 1]
```diff
@@ -122,17 +120,35 @@ class Follower {
...
-		if ( $term ) {
-			$this->id = $term->term_id;
-			$this->slug = $term->slug;
-			$this->meta = json_decode( $term->meta );
+			$post_id = $wpdb->get_var(
+				$wpdb->prepare(
+					"SELECT ID FROM $wpdb->posts WHERE guid=%s",
+					esc_sql( $actor )
+				)
+			);
```
[the remark this answers is not in the document]
Comment (pfefferle) [reply includes/model/class-follower.php:139]: This is not called by the WebFinger endpoint. Only internal users are addressable via WebFinger and this is a follower.

File: includes/model/class-follower.php [hunk 2]
```diff
@@ -351,18 +357,19 @@ public function update() {
...
-				'slug'        => sanitize_title( $this->get_actor() ),
-				'description' => wp_json_encode( $this->get_meta() ),
-			)
+		$args = array(
+			'ID'            => $this->id,
+			'guid'          => $this->actor,
+			'post_title'    => $this->get_name(),
+			'post_author'   => 0,
+			'post_type'     => Followers::POST_TYPE,
+			'post_content'  => wp_json_encode( $this->meta ),
```
[the remark this answers is not in the document]
Comment (pfefferle) [reply includes/model/class-follower.php:366]: should we use a meta instead?

[the remark this answers is not in the document]
Comment (mattwiebe) [reply includes/model/class-follower.php:366]: We might need to, but ultimately is there any required data in that or is it just nice to have?

**Pull request merged**